packages/details/test/details.test.js

```diff
@@ -58,78 +46,109 @@ describe('vaadin-details', () => {
+      expect(contentNode.getAttribute('aria-hidden')).to.equal('false');
     });
+  });
 
-    it('should set aria-controls on toggle button', () => {
-      const idRegex = /^content-vaadin-details-\d+$/u;
-      expect(idRegex.test(toggle.getAttribute('aria-controls'))).to.be.true;
+  ['default', 'custom'].forEach((type) => {
+    const fixtures = {
+      default: `
+        <vaadin-details summary="Summary">
+          <div>Content</div>
+        </vaadin-details>
+      `,
+      custom: `
+        <vaadin-details>
+          <vaadin-details-summary slot="summary">Summary</vaadin-details-summary>
+          <div>Content</div>
+        </vaadin-details>
+      `,
+    };
+
+    let summary;
+
+    describe(`${type} summary`, () => {
+      beforeEach(() => {
+        details = fixtureSync(fixtures[type]);
+        summary = details.querySelector('[slot="summary"]');
+      });
+
+      it(`should update opened on ${type} summary click`, () => {
+        summary.click();
+        expect(details.opened).to.be.true;
+
+        summary.click();
+        expect(details.opened).to.be.false;
+      });
+
+      it(`should update opened on ${type} summary Enter`, async () => {
+        summary.focus();
+
+        await sendKeys({ press: 'Enter' });
+        expect(details.opened).to.be.true;
+
+        await sendKeys({ press: 'Enter' });
+        expect(details.opened).to.be.false;
+      });
+
+      it(`should update opened on ${type} summary Space`, async () => {
+        summary.focus();
+
+        await sendKeys({ press: 'Space' });
+        expect(details.opened).to.be.true;
+
+        await sendKeys({ press: 'Space' });
+        expect(details.opened).to.be.false;
+      });
+
+      it(`should update opened on ${type} summary Arrow Down`, async () => {
+        summary.focus();
+        await sendKeys({ press: 'ArrowDown' });
+        expect(details.opened).to.be.false;
+      });
+
+      it(`should fire opened-changed event on ${type} summary click`, () => {
+        const spy = sinon.spy();
+        details.addEventListener('opened-changed', spy);
+        summary.click();
+        expect(spy.calledOnce).to.be.true;
+      });
+
+      it(`should update aria-expanded on ${type} summary click`, () => {
+        summary.click();
+        expect(summary.getAttribute('aria-expanded')).to.equal('true');
+
+        summary.click();
+        expect(summary.getAttribute('aria-expanded')).to.equal('false');
+      });
+
+      it(`should set aria-controls attribute on ${type} summary`, () => {
+        const idRegex = /^content-vaadin-details-\d+$/u;
+        expect(idRegex.test(summary.getAttribute('aria-controls'))).to.be.true;
+      });
+
+      it(`should propagate disabled attribute to ${type} summary`, () => {
+        details.disabled = true;
+        expect(summary.hasAttribute('disabled')).to.be.true;
+
+        details.disabled = false;
+        expect(summary.hasAttribute('disabled')).to.be.false;
+      });
```

Comment:
**note:** This test can be replaced with a snapshot test at some point.

Reply:
Will do it later, thanks.